src/main/java/io/cdap/plugin/sfmc/connector/MarketingConnectorConfig.java

```diff
@@ -0,0 +1,275 @@
+  }
+
+  public String getAuthEndpoint() {
+    return authEndpoint;
+  }
+
+  public String getSoapEndpoint() {
+    return soapEndpoint;
+  }
+
+  /**
+   * validates all the fields which are mandatory for the connection.
+   */
+  public void validateCredentials(FailureCollector collector) {
+    if (!shouldConnect()) {
+      return;
+    }
+
+    if (Util.isNullOrEmpty(clientId)) {
+      collector.addFailure("Client ID must be specified.", null)
+              .withConfigProperty(MarketingCloudConstants.PROPERTY_CLIENT_ID);
+    }
+
+    if (Util.isNullOrEmpty(clientSecret)) {
+      collector.addFailure("Client Secret must be specified.", null)
+              .withConfigProperty(MarketingCloudConstants.PROPERTY_CLIENT_SECRET);
+    }
+
+    if (Util.isNullOrEmpty(authEndpoint)) {
+      collector.addFailure("Auth Endpoint  must be specified.", null)
+              .withConfigProperty(MarketingCloudConstants.PROPERTY_AUTH_API_ENDPOINT);
+    }
+
+    if (Util.isNullOrEmpty(soapEndpoint)) {
+      collector.addFailure("Soap Endpoint must be specified.", null)
+              .withConfigProperty(MarketingCloudConstants.PROPERTY_SOAP_API_ENDPOINT);
+    }
+
+    collector.getOrThrowException();
+    validateSalesforceConnection(collector);
+  }
+
+  public void validateSalesforceConnection(FailureCollector collector) {
+
+    try {
+      MarketingCloudClient.create(getClientId(), getClientSecret(),
+              getAuthEndpoint(),
+              getSoapEndpoint());
+    } catch (ETSdkException e) {
+      collector.addFailure("Unable to connect to Salesforce Instance.",
+                      "Ensure properties like Client ID, Client Secret, API Endpoint " +
+                              ", Soap Endpoint, Auth Endpoint are correct.")
+              .withConfigProperty(MarketingCloudConstants.PROPERTY_CLIENT_ID)
+              .withConfigProperty(MarketingCloudConstants.PROPERTY_CLIENT_SECRET)
+              .withConfigProperty(MarketingCloudConstants.PROPERTY_AUTH_API_ENDPOINT)
+              .withConfigProperty(MarketingCloudConstants.PROPERTY_SOAP_API_ENDPOINT)
+              .withStacktrace(e.getStackTrace());
+    }
+  }
+
+  /**
+   * Returns true if Salesforce can be connected to.
+   */
+  public boolean shouldConnect() {
+    return !containsMacro(MarketingCloudConstants.PROPERTY_CLIENT_ID) &&
+            !containsMacro(MarketingCloudConstants.PROPERTY_CLIENT_SECRET) &&
+            !containsMacro(MarketingCloudConstants.PROPERTY_SOAP_API_ENDPOINT) &&
+            !containsMacro(MarketingCloudConstants.PROPERTY_AUTH_API_ENDPOINT);
+  }
+
+  public Schema getSchema(SourceObject sourceObject) {
+    try {
+      MarketingCloudClient client = MarketingCloudClient.create(this.getClientId(),
+              this.getClientSecret(),
+              this.getAuthEndpoint(),
+              this.getSoapEndpoint());
+      return MarketingCloudInputFormat.getTableMetaData(sourceObject,
+              "Stores",  client).getSchema();
```

Comment:
why Stores is hardcoded here?

Reply:
okay. I have stored the dataExtensionKey "Stores" to a constant and call the variable in the argument.

Reply:
no, it should be null as we are not calling dataextension Object from here.

Reply:
Okay. Provided dataExtensionKey as null